FirebaseAuth/Sources/Swift/MultiFactor/MultiFactor.swift

```diff
@@ -66,6 +66,57 @@ import Foundation
     public func enroll(with assertion: MultiFactorAssertion,
                        displayName: String?,
                        completion: ((Error?) -> Void)?) {
+      // TODO: Refactor classes so this duplicated code isn't necessary for phone and totp.
+      if assertion.factorID == PhoneMultiFactorInfo.TOTPMultiFactorID {
+        guard let totpAssertion = assertion as? TOTPMultiFactorAssertion else {
+          fatalError("Auth Internal Error: Failed to find TOTPMultiFactorAssertion")
+        }
+        switch totpAssertion.secretOrID {
+        case .enrollmentID: fatalError("Missing secret in totpAssertion")
+        case let .secret(secret):
+          guard let user = user, let auth = user.auth else {
+            fatalError("Internal Auth error: failed to get user enrolling in MultiFactor")
+          }
+          let finalizeMFATOTPRequestInfo =
+            AuthProtoFinalizeMFATOTPEnrollmentRequestInfo(sessionInfo: secret.sessionInfo,
+                                                          verificationCode: totpAssertion
+                                                            .oneTimePassword)
+          let request = FinalizeMFAEnrollmentRequest(idToken: self.user?.rawAccessToken(),
+                                                     displayName: displayName,
+                                                     totpVerificationInfo: finalizeMFATOTPRequestInfo,
+                                                     requestConfiguration: user
+                                                       .requestConfiguration)
+          Task {
+            do {
+              let response = try await AuthBackend.post(with: request)
+              do {
+                let user = try await auth.completeSignIn(withAccessToken: response.idToken,
+                                                         accessTokenExpirationDate: nil,
+                                                         refreshToken: response.refreshToken,
+                                                         anonymous: false)
+                try auth.updateCurrentUser(user, byForce: false, savingToDisk: true)
+                if let completion {
+                  DispatchQueue.main.async {
+                    completion(nil)
+                  }
+                }
+              } catch {
+                DispatchQueue.main.async {
```

Comment:
(No need to change this, just curious) Would the `Task`-based equivalent to `DispatchQueue.main.async` be the following?
```
Task.detached { @MainActor in
  if let completion {
    completion(error)
  }
}
```

Reply:
I think so, but not positive.